_data-prepper/common-use-cases/event-aggregation.md

```diff
@@ -0,0 +1,135 @@
+{ "log": "127.0.0.1 192.168.0.1 80", "bytes": 1000 }
+{ "log": "127.0.0.1 192.168.0.1 80" "http_verb": "GET" }
+```
+{% include copy-curl.html %}
+
+he `grok` processor will extract keys such that the log events look like the following. These events now have the data that the `aggregate` processor will need for the `identification_keys`.
+
+```json
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "port": 80, "status": 200 }
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "port": 80, "bytes": 1000 }
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "port": 80, "http_verb": "GET" }
+```
+{% include copy-curl.html %}
+
+When the group finishes 30 seconds after when the first log is received by the `aggregate` processor, the following aggregated log is written to the sink:
+
+```json
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "port": 80, "status": 200, "bytes": 1000, "http_verb": "GET" }
+```
+{% include copy-curl.html %}
+
+## Removing duplicates
+
+You can remove duplicate entries by deriving keys from incoming events and specifying the `remove_duplicates` option for the `aggregate` processor. This action immediately processes the first event for a group, and drops all following events in that group.
+
+In the following example, the first event is processed with the identification keys `sourceIp` and `destinationIp`:
+
+```json
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200 }
+```
+{% include copy-curl.html %}
+
+The pipeline will then drop the following event because it has the same keys:
+
+```json
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 1000 }
+```
+{% include copy-curl.html %}
+
+The pipeline processes this event and creates a new group because the `sourceIp` is different:
+
+```json
+{ "sourceIp": "127.0.0.2", "destinationIp": "192.168.0.1", "bytes": 1000 }
+```
+{% include copy-curl.html %}
+
+## Log aggregation and conditional routing
+
+You can use multiple plugins to combine log aggregation with conditional routing. In this example, the sub-pipeline `log-aggregate-pipeline` receives logs by using an HTTP client like FluentBit and extracts important values from the logs by matching the value in the `log` key against the common Apache log pattern.
+
+Two of the values the sub-pipeline extracts from the logs with a grok pattern include `response` and `clientip`. The `aggregate` processor then uses the `clientip` value, along with the `remove_duplicates` option, to drop any logs that contain a `clientip` that has already been processed within the given `group_duration`.
+
+Three routes, or conditional statements, exist in the pipeline. These routes separate the value of the response into 2xx/3xx, 4xx, and 5xx responses. Logs with a 2xx and 3xx status are sent to the aggregated_2xx_3xx index, logs with a 4xx status are sent to the aggregated_4xx index, and logs with a 5xx status are sent to the aggregated_5xx index.
```

Comment:
Second sentence: Confirm that it shouldn't be "2xx or 3xx status". Should the index names be in code font?

Reply:
@natebower  is correct. These are logs that have either a 2xx or 3xx status.

Reply:
@vagimeli , We want to change this:

> Logs with a `2xx` and `3xx` status

to:

> Logs with a `2xx` or `3xx` status